### PR TITLE
nixos/nixops-dns: init

### PR DESCRIPTION
**Pull request by sorki**
###### Motivation for this change

More awesome NixOps experience.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by sorki** [comment]
CC @kamilchm, Requires #34510

File: nixos/modules/services/networking/nixops-dns.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  pkg = pkgs.nixops-dns;
+  cfg = config.services.nixops-dns;
+in
+
+{
+  options = {
```
Comment (kamilchm) [inline nixos/modules/services/networking/nixops-dns.nix]: One more thing that needs to be considered - nixops database location. Current nixops-dns implementation will always look into current user home dir https://github.com/kamilchm/nixops-dns/blob/master/main.go#L23
Do you have an idea how to make it usable as system service?

Comment (sorki) [reply nixos/modules/services/networking/nixops-dns.nix]: I've tried to address this via user option (The user the nixops-dns daemon should run as.) which propagates to serviceConfig `User = cfg.user;` bellow. Not sure about the group tho.

**Review by kamilchm — APPROVED** [review]

File: nixos/modules/services/networking/nixops-dns.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    services.dnsmasq = {
+      enable = true;
+      resolveLocalQueries = true;
+      servers = [
+        "/${cfg.domain}/127.0.0.1#5300"
+      ];
+      extraConfig = ''
+        bind-interfaces
+        listen-address=127.0.0.1
```
Comment (Mic92) [inline nixos/modules/services/networking/nixops-dns.nix]: Would it not make more sense to set `listenAddress` to port 53 and give the service `cap_net_bind_service` capability to bind privileged ports? You can take a look at the caddy service on how to do this.

Comment (sorki) [reply nixos/modules/services/networking/nixops-dns.nix]: To be able to still resolve internet domains via dnsmasq while only `*.ops` are forwarded to `nixops-dns`

Comment (Mic92) [reply nixos/modules/services/networking/nixops-dns.nix]: There should be an option to disable dnsmasq in the nixops-dns module to make it less opinionated. Also it is a sane default some other people maybe prefer to run `unbound` or `knot-resolver` instead.

Comment (joachifm) [reply nixos/modules/services/networking/nixops-dns.nix]: @Mic92 do you consider this a blocker or can it be added/iterated upon later?

Comment (sorki) [reply nixos/modules/services/networking/nixops-dns.nix]: I'll update this to make dnsmasq setup an option (enabled by default).

Comment (sorki) [reply nixos/modules/services/networking/nixops-dns.nix]: Done.

**Pull request merged**